node/impl/full/eth.go

```diff
@@ -341,12 +341,29 @@ func (a *EthModule) EthGetBlockByHash(ctx context.Context, blkHash ethtypes.EthH
 	return blk, nil
 }
 
-func (a *EthModule) EthGetBlockByNumber(ctx context.Context, blkParam string, fullTxInfo bool) (ethtypes.EthBlock, error) {
+func (a *EthModule) EthGetBlockByNumber(ctx context.Context, blkParam string, fullTxInfo bool) (*ethtypes.EthBlock, error) {
+	// Get the tipset for the specified block number
 	ts, err := getTipsetByBlockNumber(ctx, a.Chain, blkParam, true)
 	if err != nil {
-		return ethtypes.EthBlock{}, err
+		if err == ErrNullRound {
+			// Return nil for null rounds
+			return nil, nil
+		}
+		return nil, xerrors.Errorf("failed to get tipset: %w", err)
+	}
+
+	// Check if the tipset is valid
+	if ts == nil || ts.Height() == 0 || len(ts.Blocks()) == 0 {
```

Comment:
why do we need these checks ?

Reply:
Let's get rid of these.

Reply:
- `ts == nil`: Ensures the tipset exists.
- `ts.Height() == 0`: Checks if it's the genesis block.
- `len(ts.Blocks()) == 0`: Ensures the tipset contains at least one block.

Reply:
Understood